PhysicsTools/PatUtils/plugins/L1PrefiringWeightProducer.cc

```diff
@@ -0,0 +1,438 @@
+  TF1* parametrization0p55To0p83_;
+  TF1* parametrization0p83To1p24_;
+  TF1* parametrization1p24To1p4_;
+  TF1* parametrization1p4To1p6_;
+  TF1* parametrization1p6To1p8_;
+  TF1* parametrization1p8To2p1_;
+  TF1* parametrization2p1To2p25_;
+  TF1* parametrization2p25To2p4_;
+  TF1* parametrizationHotSpot_;
+
+  TH2F* h_prefmap_photon_;
+  TH2F* h_prefmap_jet_;
+  const std::string dataeraEcal_;
+  const std::string dataeraMuon_;
+  const bool useEMpt_;
+  const double prefiringRateSystUncEcal_;
+  const double prefiringRateSystUncMuon_;
+  const double jetMaxMuonFraction_;
+  const bool skipwarnings_;
+  bool missingInputEcal_;
+  bool missingInputMuon_;
+};
+
+L1PrefiringWeightProducer::L1PrefiringWeightProducer(const edm::ParameterSet& iConfig)
+    : photons_token_(consumes<std::vector<pat::Photon> >(iConfig.getParameter<edm::InputTag>("ThePhotons"))),
+      jets_token_(consumes<std::vector<pat::Jet> >(iConfig.getParameter<edm::InputTag>("TheJets"))),
+      muon_token_(consumes<std::vector<pat::Muon> >(iConfig.getParameter<edm::InputTag>("TheMuons"))),
+      nonPrefiringProbToken_(produces<double>("nonPrefiringProb")),
+      nonPrefiringProbUpToken_(produces<double>("nonPrefiringProbUp")),
+      nonPrefiringProbDownToken_(produces<double>("nonPrefiringProbDown")),
+      nonPrefiringProbJetToken_(produces<double>("nonPrefiringProbJet")),
+      nonPrefiringProbJetUpToken_(produces<double>("nonPrefiringProbJetUp")),
+      nonPrefiringProbJetDownToken_(produces<double>("nonPrefiringProbJetDown")),
+      nonPrefiringProbPhotonToken_(produces<double>("nonPrefiringProbPhoton")),
+      nonPrefiringProbPhotonUpToken_(produces<double>("nonPrefiringProbPhotonUp")),
+      nonPrefiringProbPhotonDownToken_(produces<double>("nonPrefiringProbPhotonDown")),
+      nonPrefiringProbMuonToken_(produces<double>("nonPrefiringProbMuon")),
+      nonPrefiringProbMuonUpToken_(produces<double>("nonPrefiringProbMuonUp")),
+      nonPrefiringProbMuonDownToken_(produces<double>("nonPrefiringProbMuonDown")),
+      dataeraEcal_(iConfig.getParameter<std::string>("DataEraECAL")),
+      dataeraMuon_(iConfig.getParameter<std::string>("DataEraMuon")),
+      useEMpt_(iConfig.getParameter<bool>("UseJetEMPt")),
+      prefiringRateSystUncEcal_(iConfig.getParameter<double>("PrefiringRateSystematicUnctyECAL")),
+      prefiringRateSystUncMuon_(iConfig.getParameter<double>("PrefiringRateSystematicUnctyMuon")),
+      jetMaxMuonFraction_(iConfig.getParameter<double>("JetMaxMuonFraction")),
+      skipwarnings_(iConfig.getUntrackedParameter<bool>("SkipWarnings")) {
+  missingInputEcal_ = false;
+  missingInputMuon_ = false;
+
+  std::string fname = iConfig.getParameter<std::string>("L1Maps");
+  edm::FileInPath mapsfilepath("PhysicsTools/PatUtils/data/" + fname);
+  file_prefiringmaps_ = std::make_unique<TFile>(mapsfilepath.fullPath().c_str(), "read");
+  if (file_prefiringmaps_ == nullptr) {
+    missingInputEcal_ = true;
+    if (!skipwarnings_)
```

Comment:
I have the impression that most of those warnings are really worrisome ones, and should always get reported, not only when someone remembers to set the `skipwarnings_` flag to `false` in the config file.

Since you have moved from previous `cout` to the `MessageLogger,` I'd profit of it and modulate the warnings according to their severity, getting rid of the `skipwarnings_` parameter. For example, here if an input file with maps/parameters is missing this is a real error that will affect all following calculations, and should be notified with a `LogError`. At a first glance. most of the "warnings" that you control with `skipwarnings_` could be promoted to `LogError` (i.e. all those in the class construcor); the less relevant ones, could be either left as `LogWarning` (if expected to occur at low frequency) or `LogDebug` (if you want to access them only in debug mode: I would do so for the one at [L406](https://github.com/cms-sw/cmssw/pull/33758/files#diff-4052fd6f31ddfc7812747e0a524c682f5344bc807fd6f55b2228e6a7bf2731f5R406))

Reply:
I agree, done.